distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Type_Refinements/Single_Value_Column.enso

```diff
@@ -0,0 +1,81 @@
+private
+
+from Standard.Base import all
+
+import project.Column.Column
+import project.Value_Type.Value_Type
+
+refine_with_single_value (column : Column) =
+    ## We treat a column as single value if it contains a single not-nothing value.
+    if _is_single_value column . not then column else case column.inferred_precise_value_type of
+        Value_Type.Integer _ ->
+            # `inferred_precise_value_type` will return Integer if the column was Float but contained integral values - e.g. [2.0]
+            was_originally_float = column.value_type.is_floating_point
+            if was_originally_float then (column : Column & Float & Integer) else (column : Column & Integer)
+        Value_Type.Float _ -> (column : Column & Float)
+        Value_Type.Char _ _ -> (column : Column & Text)
+        Value_Type.Boolean -> (column : Column & Boolean)
+        Value_Type.Date -> (column : Column & Date)
+        Value_Type.Time -> (column : Column & Time_Of_Day)
+        Value_Type.Date_Time True -> (column : Column & Date_Time)
+        Value_Type.Decimal _ scale ->
+            is_integer = scale == 0
+            if is_integer then (column : Column & Integer) else (column : Column & Decimal)
+        # Other types (e.g. Mixed) are not supported.
+        _ -> column
+
+_is_single_value column:Column -> Boolean =
+    (column.length == 1) && (column.at 0 . is_nothing . not)
+
+## PRIVATE
```

Comment:
Does having all the conversions in the same file cause any issues?

Reply:
I don't think so but I can also move it to a separate file if we prefer.

Reply:
Moved.